server/src/test/java/org/elasticsearch/search/SearchCancellationTests.java

```diff
@@ -79,14 +81,21 @@ public static void cleanup() throws IOException {
         reader = null;
     }
 
-    public void testAddingCancellationActions() throws IOException {
-        ContextIndexSearcher searcher = new ContextIndexSearcher(
+    private ContextIndexSearcher createSearcher() throws IOException {
+        // TODO maybe randomize setting executor here
+        return new ContextIndexSearcher(
             reader,
             IndexSearcher.getDefaultSimilarity(),
             IndexSearcher.getDefaultQueryCache(),
             IndexSearcher.getDefaultQueryCachingPolicy(),
-            true
+            1,
+            true,
+            null
```

Comment:
I am thinking that we should have a newContextSearch method in ESTestCase, similar to LuceneTestCase#newSearcher , and take care of the randomization there. This centralizes tha randomized aspects, and also removes the need to fiddle with the executor in each test.